sdk/servicebus/service-bus/test/stress-test-track-2/scenarioRenewMessageLock.ts

```diff
@@ -0,0 +1,108 @@
+  delayBetweenSendsInMs?: number;
+  totalNumberOfMessagesToSend?: number;
+  completeMessageAfterDuration?: boolean;
+}
+
+function sanitizeOptions(
+  options: ScenarioRenewMessageLockOptions
+): Required<ScenarioRenewMessageLockOptions> {
+  return {
+    testDurationInMs: options.testDurationInMs || 60 * 60 * 1000, // Default = 60 minutes
+    receiveBatchMaxMessageCount: options.receiveBatchMaxMessageCount || 10,
+    receiveBatchMaxWaitTimeInMs: options.receiveBatchMaxWaitTimeInMs || 10000,
+    delayBetweenReceivesInMs: options.delayBetweenReceivesInMs || 0,
+    numberOfMessagesPerSend: options.numberOfMessagesPerSend || 1,
+    delayBetweenSendsInMs: options.delayBetweenSendsInMs || 0,
+    totalNumberOfMessagesToSend: options.totalNumberOfMessagesToSend || Infinity,
+    completeMessageAfterDuration: options.completeMessageAfterDuration || true
+  };
+}
+
+// TODO: max lock renewal duration to be 70% of testDuration instead of 100%
+// TODO: stop sending messages after a 70% of test duration
+// TODO: Upon ending max lock renewal duration, pass an option to complete/ignore the message
+export async function main() {
+  const {
+    testDurationInMs,
+    receiveBatchMaxMessageCount,
+    receiveBatchMaxWaitTimeInMs,
+    delayBetweenReceivesInMs,
+    numberOfMessagesPerSend,
+    delayBetweenSendsInMs,
+    totalNumberOfMessagesToSend,
+    completeMessageAfterDuration
+  } = sanitizeOptions(parsedArgs<ScenarioRenewMessageLockOptions>(process.argv));
+
+  const testDurationForSendInMs = testDurationInMs * 0.7;
+  // TODO: Randomize the duration to renew locks
+  const testDurationForLockRenewalInMs = testDurationInMs * 0.5;
+
+  const startedAt = new Date();
+
+  const stressBase = new SBStressTestsBase({
+    snapshotFocus: ["send-info", "receive-info", "message-lock-renewal-info"]
+  });
+  const sbClient = new ServiceBusClient(connectionString);
+
+  await stressBase.init();
+  const sender = sbClient.createSender(stressBase.queueName);
+  const receiver = sbClient.createReceiver(stressBase.queueName, { receiveMode: "peekLock" });
+
+  async function sendMessages() {
+    let elapsedTime = new Date().valueOf() - startedAt.valueOf();
+    while (
+      elapsedTime < testDurationForSendInMs &&
+      stressBase.messagesSent.length < totalNumberOfMessagesToSend
+    ) {
+      await stressBase.sendMessages([sender], numberOfMessagesPerSend);
+      elapsedTime = new Date().valueOf() - startedAt.valueOf();
+      await delay(delayBetweenSendsInMs);
+    }
+  }
+
+  async function receiveMessages() {
+    let elapsedTime = new Date().valueOf() - startedAt.valueOf();
+    while (elapsedTime < testDurationInMs) {
+      const messages = await stressBase.receiveMessages(
+        receiver,
+        receiveBatchMaxMessageCount,
+        receiveBatchMaxWaitTimeInMs
+      );
+      elapsedTime = new Date().valueOf() - startedAt.valueOf();
+      messages.map((msg) =>
+        stressBase.renewMessageLockUntil(
```

Comment:
It is not clear to me what our expectation here is...
The `renewMessageLockUntil` will keep renewing the lock... what are we looking for? That no error occurs?
Where will the error surface?

And when does the program end? We are awaiting on receiveMessages() but not really for the renewMessageLockUntil to finish?

Reply:
The time duration for `renewMessageLockUntil` is less than the testDuration always, and hence would end before, and lock renewals are being done through setTimeouts.

The intention behind the test is to have many messages for which we'd be renewing the locks, note down the errors if any, observe the limits.. as in number of messages can we handle, etc.